quarkus-kafka-server-extension/deployment/src/main/java/com/ozangunalp/kafka/server/extension/deployment/KafkaServerExtensionProcessor.java

```diff
@@ -127,9 +127,12 @@ void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             "org.apache.kafka.common.security.oauthbearer.internals.expiring.ExpiringCredentialRefreshingLogin"));
         producer.produce(new RuntimeInitializedClassBuildItem(
             "org.apache.kafka.common.security.kerberos.KerberosLogin"));
-        
+        producer.produce(new RuntimeInitializedClassBuildItem(
+            "org.apache.kafka.storage.internals.log.LogSegment"));
```

Comment:
I had to do this because LogSegment has a static initializer.  I tried @Deleting the field used by the static initializer, but the native image failed to build.

Reply:
I should be able to get rid of the static block in LogSegment using bytecode manipulation.

Reply:
@k-wall I pushed an additional commit to your remote